uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/SicknessRequestTest.php

```diff
@@ -0,0 +1,191 @@
+use CRM_HRLeaveAndAbsences_BAO_SicknessRequest as SicknessRequest;
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
+
+/**
+ * Class CRM_HRLeaveAndAbsences_BAO_SicknessRequestTest
+ *
+ * @group headless
+ */
+class CRM_HRLeaveAndAbsences_BAO_SicknessRequestTest extends BaseHeadlessTest {
+
+  use CRM_HRLeaveAndAbsences_LeaveRequestHelpersTrait;
+  use CRM_HRLeaveAndAbsences_LeaveBalanceChangeHelpersTrait;
+  use CRM_HRLeaveAndAbsences_SicknessRequestHelpersTrait;
+
+  public function setUp() {
+    CRM_Core_DAO::executeQuery("SET foreign_key_checks = 0;");
+
+    $this->requiredDocumentOptions = $this->requiredDocumentOptionsBuilder();
+    $this->leaveRequestDayTypes = $this->leaveRequestDayTypeOptionsBuilder();
+  }
+
+  public function testCreateSicknessRequest() {
+    $contact = ContactFabricator::fabricate();
+    $period = AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2016-01-01'),
+      'end_date'   => CRM_Utils_Date::processDate('2016-12-31'),
+    ]);
+
+    $absenceType = AbsenceTypeFabricator::fabricate([
+      'title' => 'Type 1',
+      'allow_overuse' => 1
+    ]);
+
+    $periodEntitlement = LeavePeriodEntitlementFabricator::fabricate([
+      'type_id' => $absenceType->id,
+      'contact_id' => $contact['id'],
+      'period_id' => $period->id
+    ]);
+
+    $this->createLeaveBalanceChange($periodEntitlement->id, 3);
+    $periodStartDate = date('2016-01-01');
+    $title = 'Job Title';
+
+    HRJobContractFabricator::fabricate([
+      'contact_id' => $contact['id']
+    ],
+    [
+      'period_start_date' => $periodStartDate,
+      'title' => $title
+    ]);
+
+    $workPattern = WorkPatternFabricator::fabricateWithA40HourWorkWeek();
+    ContactWorkPatternFabricator::fabricate([
+      'contact_id' => $contact['id'],
+      'pattern_id' => $workPattern->id
+    ]);
+
+    $fromDate = new DateTime("2016-11-14");
+    $toDate = new DateTime("2016-11-17");
+    $fromType = $this->leaveRequestDayTypes['All Day']['id'];
+    $toType = $this->leaveRequestDayTypes['All Day']['id'];
+
+    //four working days which will create a balance change of 4
+    $sicknessReasons = array_flip(SicknessRequest::buildOptions('reason'));
+
+    $sicknessRequest = SicknessRequest::create([
+      'type_id' => $absenceType->id,
+      'contact_id' => $contact['id'],
+      'status_id' => 1,
+      'from_date' => $fromDate->format('YmdHis'),
+      'from_date_type' => $fromType,
+      'to_date' => $toDate->format('YmdHis'),
+      'to_date_type' => $toType,
+      'reason' => $sicknessReasons['Appointment'],
+      'required_documents' => $this->requiredDocumentOptions['Self certification form required']['value']. ',' . $this->requiredDocumentOptions['Back to work interview required']['value'],
+    ]);
+
+    $this->assertInstanceOf(SicknessRequest::class, $sicknessRequest);
+    $this->assertEquals($sicknessRequest->reason, $sicknessReasons['Appointment']);
```

Comment:
I think it would be better to also add a $validate flag to the SicknessRequest.create method, which will also be passed to the LeaveRequest.create method, in order to avoid such a big and slow test.

Regarding what is being tested, I think it would be nice to check that the related LeaveRequest and LeaveRequest dates were created properly